compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -1395,6 +1398,25 @@ uint8_t* TR::X86RegRegInstruction::generateOperand(uint8_t* cursor)
    }
 
 
+// -----------------------------------------------------------------------------
+// TR::X86RegRegInstruction:: member functions
+
+uint8_t* TR::X86RegRegRegInstruction::generateOperand(uint8_t* cursor)
+   {
+   uint8_t *modRM = cursor - 1;
+   if (getOpCode().hasTargetRegisterIgnored() == 0)
+      {
+      applyTargetRegisterToModRMByte(modRM);
+      }
+   if (getOpCode().hasSourceRegisterIgnored() == 0)
+      {
+      applySourceRegisterToModRMByte(modRM);
+      }
+   applySource2ndRegisterToVEX(modRM - 2);
```

Comment:
This is forever coupling the `RegRegReg` form instruction to those that have VEX prefixes.  At the moment that's true, but I'm not sure if it always will be.

Can you guard this statement with a `if (getOpCode().info().supportsAVX())` just in case ?

Reply:
Good point, I've added an assertion to ensure the instruction that have VEX prefixes.